packages/@uppy/react/src/DashboardModal.ts

```diff
@@ -0,0 +1,275 @@
+    height: 550,
+    thumbnailWidth: 280,
+    thumbnailType: 'image/jpeg',
+    waitForThumbnailsBeforeUpload: false,
+    showLinkToFileUploadResult: false,
+    showProgressDetails: false,
+    hideUploadButton: false,
+    hideCancelButton: false,
+    hideRetryButton: false,
+    hidePauseResumeButton: false,
+    hideProgressAfterFinish: false,
+    note: null,
+    closeModalOnClickOutside: false,
+    closeAfterFinish: false,
+    disableStatusBar: false,
+    disableInformer: false,
+    disableThumbnailGenerator: false,
+    disablePageScrollWhenModalOpen: true,
+    animateOpenClose: true,
+    fileManagerSelectionType: 'files',
+    proudlyDisplayPoweredByUppy: true,
+    showSelectedFiles: true,
+    showRemoveButtonAfterComplete: false,
+    browserBackButtonClose: false,
+    theme: 'light',
+    autoOpenFileEditor: false,
+    disabled: false,
+    disableLocalFiles: false,
+
+    // extra
+    open: undefined,
+    target: undefined,
+    locale: null,
+    onRequestClose: undefined,
+  }
+
+  private container: HTMLElement
+
+  private plugin: DashboardPlugin<M, B>
+
+  componentDidMount(): void {
+    this.installPlugin()
+  }
+
+  componentDidUpdate(prevProps: DashboardModal<M, B>['props']): void {
+    const { uppy, open, onRequestClose } = this.props
+    if (prevProps.uppy !== uppy) {
+      this.uninstallPlugin(prevProps)
+      this.installPlugin()
+    } else if (nonHtmlPropsHaveChanged(this.props, prevProps)) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars,no-shadow
+      const { uppy, ...options } = {
+        ...this.props,
+        onRequestCloseModal: onRequestClose,
+      }
+      this.plugin.setOptions(options)
+    }
+    if (prevProps.open && !open) {
+      this.plugin.closeModal()
+    } else if (!prevProps.open && open) {
+      this.plugin.openModal()
+    }
+  }
+
+  componentWillUnmount(): void {
+    this.uninstallPlugin()
+  }
+
+  installPlugin(): void {
+    const {
+      uppy,
+      target,
+      open,
+      onRequestClose,
+      closeModalOnClickOutside,
+      disablePageScrollWhenModalOpen,
+      inline,
+      plugins, // eslint-disable-line no-shadow
+      width,
+      height,
+      showProgressDetails,
+      note,
+      metaFields, // eslint-disable-line no-shadow
+      proudlyDisplayPoweredByUppy,
+      autoOpenFileEditor,
+      animateOpenClose,
+      browserBackButtonClose,
+      closeAfterFinish,
+      disableStatusBar,
+      disableInformer,
+      disableThumbnailGenerator,
+      disableLocalFiles,
+      disabled,
+      hideCancelButton,
+      hidePauseResumeButton,
+      hideProgressAfterFinish,
+      hideRetryButton,
+      hideUploadButton,
+      showLinkToFileUploadResult,
+      showRemoveButtonAfterComplete,
+      showSelectedFiles,
+      waitForThumbnailsBeforeUpload,
+      fileManagerSelectionType,
+      theme,
+      thumbnailType,
+      thumbnailWidth,
+      locale, // eslint-disable-line no-shadow
+    } = this.props
```

Comment:
I think this is one example where prop speading is actually fine. We do want to pass on _everything_ as it's the top level component and make this more readable.

Reply:
It was already like that before this PR, I prefer to keep it as is.

Reply:
@lakesare is removing them here: https://github.com/transloadit/uppy/pull/5001.

I think I agree with her that this is a good example where do we want spreading.